dotCMS/src/main/java/com/dotcms/cluster/business/HazelcastUtil.java

```diff
@@ -1,74 +1,198 @@
             Logger.error(HazelcastUtil.class, "Could not initialize Hazelcast Member", e);
         }
-        return _memberInstance;
+        return _memberInstances.get(instanceType);
     }
 
-    public void shutdown(){
-        if (_memberInstance == null) {
+    public void shutdown(HazelcastInstanceType instanceType){
+        if (_memberInstances.get(instanceType) != null) {
             synchronized (syncMe) {
-                if ( _memberInstance == null) {
-                    _memberInstance.shutdown();
-                    _memberInstance = null;
+                if (_memberInstances.get(instanceType) != null) {
+                	_memberInstances.get(instanceType).shutdown();
+                	_memberInstances.remove(instanceType);
                 }
             }
         }
     }
 
-    private void initMember () {
+    public void shutdown() {
+        for(HazelcastInstanceType instanceType : HazelcastInstanceType.values()){
+        	shutdown(instanceType);        	
+        }
+    }
+
+    private void initMember(HazelcastInstanceType instanceType, boolean reInitialize) {
 
-        if (_memberInstance == null) {
+        if (_memberInstances.get(instanceType) == null || reInitialize) {
             long start = System.currentTimeMillis();
             synchronized (syncMe) {
-                if ( _memberInstance == null) {
-                    if(xmlFilePath == null){
-                        xmlFilePath = "hazelcast-embedded.xml";
+                if (_memberInstances.get(instanceType) == null || reInitialize) {
+                    Logger.info(this, "Setting Up HazelCast ("+ instanceType +")");
+
+                    HazelcastInstance memberInstance = _memberInstances.remove(instanceType);
+                    if (memberInstance != null) {
+                    	try {
+                    		Thread.sleep(1000);
+                    	} catch (InterruptedException e) {
+                    		Logger.error(this, "Error waiting instance to become vacant", e);
+                    	} finally {
+                    		memberInstance.shutdown();
+                    	}
                     }
-                    Logger.info(this, "Setting Up HazelCast");
-                    InputStream is = null;
-                    try {
-                        is = getClass().getClassLoader().getResourceAsStream(xmlFilePath);
-                        XmlConfigBuilder builder = new XmlConfigBuilder(is);
-                        _memberInstance = Hazelcast.newHazelcastInstance(builder.build());
-                    } finally {
-                        if (is != null) {
-                            try {
-                                is.close();
-                            }catch (Exception e){
-                                Logger.error(Hazelcast.class, "Unable to close inputstream for Hazel xml", e);
-                            }
-                        }
+
+                    if (instanceType == HazelcastInstanceType.EMBEDDED) {
```

Comment:
@agomez-dotcms can we move this code into a Factory? HazelCastInstanceFactory, and have a getInstance(String xmlPath) method that has the if else and proper creation?